src/ERC6538Registry.sol

```diff
@@ -24,22 +68,177 @@ contract ERC6538Registry is IERC6538Registry {
+  /// @param hash The digest of message data.
+  /// @param signature The signature provided by the registrant.
+  function isValidERC1271SignatureNow(address signer, bytes32 hash, bytes memory signature)
+    internal
+    view
+    returns (bool)
+  {
+    (bool success, bytes memory result) =
+      signer.staticcall(abi.encodeCall(IERC1271.isValidSignature, (hash, signature)));
+    return (
+      success && result.length >= 32
+        && abi.decode(result, (bytes32)) == bytes32(IERC1271.isValidSignature.selector)
+    );
+  }
+
+  /// @dev The following code is from OpenZeppelin's `ECDSA.sol` file. Permalink:
+  /// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e70a0118ef10773457f670671baefad2c5ea610d/contracts/utils/cryptography/ECDSA.sol
+  /// @param hash The digest of message data.
+  /// @param signature The signature provided by the registrant.
+  function tryRecover(bytes32 hash, bytes memory signature)
+    internal
+    pure
+    returns (address, RecoverError, bytes32)
+  {
+    if (signature.length == 65) {
+      bytes32 r;
+      bytes32 s;
+      uint8 v;
+      assembly {
+        r := mload(add(signature, 0x20))
+        s := mload(add(signature, 0x40))
+        v := byte(0, mload(add(signature, 0x60)))
+      }
+      return tryRecover(hash, v, r, s);
+    } else {
+      return (address(0), RecoverError.InvalidSignatureLength, bytes32(signature.length));
+    }
+  }
+
+  /// @notice Recover the signer's address using `v`, `r` and `s` signature fields.
+  /// @dev The following code is from OpenZeppelin's `ECDSA.sol` file. Permalink:
+  /// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e70a0118ef10773457f670671baefad2c5ea610d/contracts/utils/cryptography/ECDSA.sol
+  function tryRecover(bytes32 hash, uint8 v, bytes32 r, bytes32 s)
+    internal
+    pure
+    returns (address, RecoverError, bytes32)
+  {
+    /// EIP-2 still allows signature malleability for ecrecover(). Remove this possibility and make
+    /// the signature unique. Appendix F in the Ethereum Yellow paper
+    /// (https://ethereum.github.io/yellowpaper/paper.pdf), defines the valid range for s in (301):
+    /// 0 < s < secp256k1n ÷ 2 + 1, and for v in (302): v ∈ {27, 28}. Most signatures from
+    /// current libraries generate a unique signature with an s-value in the lower half order.
+
+    /// If your library generates malleable signatures, such as s-values in the upper range,
+    /// calculate a new s-value with
+    /// 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141 - s1 and flip v from 27
+    /// to 28 or vice versa. If your library also generates signatures with 0/1 for v instead 27/28,
+    /// add 27 to v to accept these malleable signatures as well.
+    if (uint256(s) > 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0) {
+      return (address(0), RecoverError.InvalidSignatureS, s);
+    }
+
+    /// If the signature is valid (and not malleable), return the signer address
```

Comment:
nit, these comments use triple slashes but should be double slash

Reply:
oh actually this is copied from OZ right? maybe leave it then to minimize the diff, up to you all :)

Reply:
Thanks for your patience, I've updated the comments to use `//` in https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/9372812c18bdef57e31ca84b7a0490e0d3089979